lib/ae_mdw/channels.ex

```diff
@@ -0,0 +1,52 @@
+defmodule AeMdw.Channels do
+  @moduledoc """
+  Main channels module.
+  """
+
+  alias AeMdw.Collection
+  alias AeMdw.Db.ChannelCloseMutation
+  alias AeMdw.Db.ChannelOpenMutation
+  alias AeMdw.Db.ChannelSpendMutation
+  alias AeMdw.Db.Model
+  alias AeMdw.Db.State
+  alias AeMdw.Node
+  alias AeMdw.Txs
+
+  require Model
+
+  @typep state() :: State.t()
+  @type closing_type() ::
+          :channel_close_solo_tx
+          | :channel_close_mutual_tx
+          | :channel_settle_tx
+
+  @spec close_mutation(closing_type(), Node.aetx()) :: ChannelCloseMutation.t()
+  def close_mutation(tx_type, tx), do: ChannelCloseMutation.new(tx_type, tx)
+
+  @spec open_mutation(Node.aetx()) :: ChannelOpenMutation.t()
+  def open_mutation(tx), do: ChannelOpenMutation.new(tx)
+
+  @spec deposit_mutation(Node.aetx()) :: ChannelSpendMutation.t()
+  def deposit_mutation(tx), do: ChannelSpendMutation.new(:aesc_deposit_tx.amount(tx))
+
+  @spec withdraw_mutation(Node.aetx()) :: ChannelSpendMutation.t()
+  def withdraw_mutation(tx), do: ChannelSpendMutation.new(-:aesc_withdraw_tx.amount(tx))
```

Comment:
It's neat but personally I would call the `new` directly avoiding the indirection as they are used only once.

Reply:
The purpose of this was to encapsulate the :aesc module inside the `Channels` module so that the `Transaction` module didn't have to deal with them directly